When cross compiling, the `gethost` binary needs to be built for the build
host and there is a specific `CC_FOR_GETHOST` variable that controls that.
It unfortunately always defaults to `cc`.

diff -wpruN --no-dereference '--exclude=*.orig' a~/configure.ac a/configure.ac
--- a~/configure.ac	1970-01-01 00:00:00
+++ a/configure.ac	1970-01-01 00:00:00
@@ -357,7 +357,7 @@ AC_PATH_PROG([GENCAT], [gencat])
 
 dnl Require build CC to create gethost helper when cross building
 AS_IF([test "x${cross_compiling}" = xyes],
-      [CC_FOR_GETHOST="cc"],
+      [CC_FOR_GETHOST=${CC_FOR_GETHOST:-"cc"}],
       [CC_FOR_GETHOST="\$(CC)"])
 AC_SUBST(CC_FOR_GETHOST)
 
